### PR TITLE
screencast: check if a valid format was received from wlr_screencopy

### PR DESCRIPTION
**Pull request by columbarius**
When a screencast is started we do a roundtrip on get the offered
formats from wlr_screencopy. This roundtrip can fail [1]. In this case
the formats won't be initialized and we will just error out of the
screencast session.

[1] https://github.com/emersion/xdg-desktop-portal-wlr/issues/214

### CODE REVIEW

File: src/screencast/screencast.c [hunk 1]
```diff
@@ -167,6 +168,13 @@ static int start_screencast(struct xdpw_screencast_instance *cast) {
 	wl_display_dispatch(cast->ctx->state->wl_display);
 	wl_display_roundtrip(cast->ctx->state->wl_display);
 
+	if (cast->screencopy_frame_info[WL_SHM].format == DRM_FORMAT_INVALID ||
```
Comment (emersion) [inline src/screencast/screencast.c:171]: The SHM format should not be a DRM format? See `enum wl_shm_format`. 0 there means `WL_SHM_FORMAT_XRGB8888`, and `DRM_FORMAT_INVALID` also happens to be 0.

**Comment by emersion** [comment]
Maybe it would be easier to set a bool field when receiving the `buffer_done` event, and check that field after the roundtrip?

File: src/screencast/screencast.c (hunk 1 above)
Comment (columbarius) [reply src/screencast/screencast.c:171]: We are using drm_format for everything internally, so this shouldn't be an issue. (see https://github.com/emersion/xdg-desktop-portal-wlr/blob/master/src/screencast/wlr_screencast.c#L106)

Comment (emersion) [reply src/screencast/screencast.c:171]: Oh okay that makes sense then

**Comment by columbarius** [comment]
I'd rather not use a flag for each state of setup we reached. Since we use drm_format internally for shm, the 0 shouldn't be an issue. I thought about making it possible for compositors to drop shm buffer screencopy in the future (signaled with an invalid shm_format, if there is one) and this could work with checking the format, but I saw no reason to make a more detailed check here. I assume that failing screencopy, because the compositor can't find a usable format (while we are fine with any at that point) should not be common on released swas/wlroot versions.

**Pull request merged**